### PR TITLE
"fixed" sampling with old <eos> method

### PR DESCRIPTION
**Pull request by lucidrains**
@sdtblck yet another try

### CODE REVIEW

**Comment by lucidrains** [comment]
@sdtblck if this still doesn't work, we'll give the non-cached version on the same branch a try

**Comment by lucidrains** [comment]
@sdtblck hmm, i don't see the logits mask being used at all, so it's likely going to be still broken with this old way of doing things
